sklego/preprocessing.py

```diff
@@ -206,3 +206,55 @@ def _check_X_for_type(X):
         """Checks if input of the Selector is of the required dtype"""
         if not isinstance(X, pd.DataFrame):
             raise TypeError("Provided variable X is not of type pandas.DataFrame")
+
+
+class OrthogonalTransformer(BaseEstimator, TransformerMixin):
+    """
+    Transform the columns of a dataframe or numpy array to a column orthogonal or orthonormal matrix.
+    Q, R such that X = Q*R, with Q orthogonal, from which follows Q = X*inv(R)
+    :param normalize: whether orthogonal matrix should be orthonormal as well
+    """
+
+    def __init__(self, normalize=False):
+        self.normalize = normalize
+
+    def fit(self, X, y=None):
+        """
+        Store the inverse of R of the QR decomposition of X, which can be used to calculate the orthogonal projection
+        of X. If normalization is required, also stores a vector with normalization terms
+        """
+        input_is_df = isinstance(X, pd.DataFrame)
+
+        if input_is_df:
+            X = np.array(X)
+
+        # Q, R such that X = Q*R, with Q orthogonal, from which follows Q = X*inv(R)
+        Q, R = np.linalg.qr(X)
+        self.inv_r_ = np.linalg.inv(R)
+
+        if self.normalize:
```

Comment:
I think it would be nice to always have this `normalization_vector_` but if `self.normalize` is False, just instantiate it as a vector of ones. That saves two if statements in the `transform`

Reply:
I agree, changed in the latest commit